app/models/atmosphere/virtual_machine_flavor.rb

```diff
@@ -60,6 +64,48 @@ class VirtualMachineFlavor < ActiveRecord::Base
 
     scope :active, -> { where(active: true) }
 
+    # Assumes only 1 vmf_osf for a specific os_family
+    def get_hourly_cost_for(os_family)
+      incarnation = virtual_machine_flavor_os_families.select{|vmf_osf| vmf_osf.os_family == os_family}.first
+      if incarnation.is_a?(Atmosphere::VirtualMachineFlavorOSFamily)
+        incarnation.hourly_cost
+      else
+        nil #GIGO
+      end
+    end
+
+    # Upserts a binding between this VMF and an OS family, setting hourly cost
+    def set_hourly_cost_for(os_family, cost)
+      incarnation = virtual_machine_flavor_os_families.select{|vmf_osf| vmf_osf.os_family == os_family}.first
+      if incarnation.is_a?(Atmosphere::VirtualMachineFlavorOSFamily)
+        incarnation.hourly_cost = cost
+        incarnation.save
+      else
+        incarnation = Atmosphere::VirtualMachineFlavorOSFamily.new(virtual_machine_flavor: self, os_family: os_family,
+          hourly_cost: cost)
+        incarnation.save
+      end
```

Comment:
It can be simplified as follow:

```ruby
def set_hourly_cost_for(os_family, cost)
  incarnation = virtual_machine_flavor_os_families.find_or_initialize_by(os_family: os_family)
  incarnation.hourly_cost = cost
  incarnation.save
end
```